crates/circuit/src/dag_circuit.rs

```diff
@@ -1677,14 +1673,12 @@ def _format(operand):
         let qargs = qargs.map(|q| q.value);
         let cargs = cargs.map(|c| c.value);
         let node = {
-            let qubits_id = Interner::intern(
-                &mut self.qargs_cache,
-                self.qubits.map_bits(qargs.iter().flatten())?.collect(),
-            )?;
-            let clbits_id = Interner::intern(
-                &mut self.cargs_cache,
-                self.clbits.map_bits(cargs.iter().flatten())?.collect(),
-            )?;
+            let qubits_id = self
+                .qargs_cache
+                .insert_owned(self.qubits.map_bits(qargs.iter().flatten())?.collect());
```

Comment:
This might be more straightforward than you had in mind - `<[T] as ToOwned>::Owned` is `Vec<T>`, so this is basically identical to what it was before.  It just looks shorter because it's easier to access the `insert` method, and the `Vec<T>` is a bit more obfuscated (unfortunately - I actually wanted it to be more obvious, but the only way I could think to do that was to introduce a second type parameter that had no real effect).

Reply:
Oh wow, I would not have guessed that. Thanks for clarifying!

I agree with your choice to avoid a redundant type parameter.